aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,312 @@
+  :lines: 21
+  :dedent: 12
+
+``ServiceFilterAttribute`` implements `IFilterFactory <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/IFilterFactory/index.html>`_, which exposes a single method for creating an ``IFilter`` instance. In the case of ServiceFilter, this is implemented by using the built-in service provider to load the type from DI.
+
+``TypeFilterAttribute`` is very similar to ``ServiceFilterAttribute`` (and also implements ``IFilterFactory``), but its type is not resolved directly from the DI container. Instead, it instantiates the type using an `ObjectFactory delegate <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/Extensions/DependencyInjection/ObjectFactory/index.html>`_. Because of this difference, types that are referenced using the ``TypeFilterAttribute`` do not need to be registered with the container first (but they will still have their dependencies fulfilled by the container). Also, ``TypeFilterAttribute`` can optionally accept constructor arguments for the type in question. The following example demonstrates how to pass arguments to a type using ``TypeFilterAttribute``:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Controllers/HomeController.cs
+  :language: c#
+  :emphasize-lines: 1-2
+  :lines: 33-39
+  :dedent: 8
+
+If you have a simple filter that doesn't require any arguments, but which has constructor dependencies that need to be filled by DI, you can inherit from ``TypeFilterAttribute``, allowing you to use your own named attribute on classes and methods (instead of ``[TypeFilter(typeof(FilterType))]``). The following filter shows how this can be implemented:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/SampleLoggingActionFilterAttribute.cs
+  :language: c#
+  :emphasize-lines: 1, 3, 7
+  :lines: 8-38
+  :dedent: 4
+
+This filter can be applied to classes or methods using the ``[SampleLoggingActionFilter]`` syntax.
+
+.. note:: Some of the filters shown in these examples implement logging in order to demonstrate when, where, or how they are executing. You should avoid creating and using filters purely for logging purposes, since the :doc:`built-in framework logging features </fundamentals/logging>` should already provide what you need to see how MVC is working under the covers. If you're going to add logging to your filters, it should focus on business domain concerns or behavior specific to your filter, rather than MVC actions or other framework events.
+
+``IFilterFactory`` actually implements ``IFilter``, so anywhere in the filter pipeline an ``IFilterFactory`` can be used in place of an ``IFilter`` instance. When it comes time for a filter to run, it is cast to an ``IFilterFactory``. If that cast succeeds, then the ``CreateInstance`` method is called to create the actual ``IFilter`` that will be used. This provides a very flexible design, since the precise filter pipeline does not need to be set in stone when the application starts.
```

Comment:
Clarified - let me know if you still would change it. The flexibility is the same flexibility any Factory pattern offers, which is that in place of a specific instance of a type, you provide a factory and the actual instance is provided at runtime by the factory (using whatever logic the factory implements).